### ISSUE TITLE
aws-servicediscovery: IHttpNamespace doesn't include createService

### ISSUE THREAD

**Issue by Danno040** [aws/aws-cdk#21135]
### Describe the bug

If you try to import any Cloud Map Namespaces using the `fromHttpNamespaceAttributes` method, you cannot call `createService` on the returned object because `IHttpNamespace` doesn't include it.

This is also true of `IPrivateDnsNamespace` and `IPublicDnsNamespace`.

### Expected Behavior

The following example should work:

```typescript
class ImportedNamespaceStack extends Stack {
  constructor(scope: Construct, name: string, props: Props) {
    const importedNamespace = HttpNamespace.fromHttpNamespaceAttributes(this, 'imported-namespace', {
      namespaceName: 'imported',
      namespaceId: 'ns-iwasimported',
      namespaceArn: 'arn:aws:servicediscovery:us-east-1:123456789012:namespace/ns-iwasimported',
    })
   importedNamespace.createService('my-cool-service', { name: 'my-cool-service' })
  }
}
```

### Current Behavior

It does not compile because `IHttpNamespace`  has no method called `createService`.

### Reproduction Steps

See example in expected behavior

### Possible Solution

Expand the interface to include the method:

```typescript
export interface IHttpNamespace extends INamespace {
  /**
   * Creates a service within the namespace
   */
  createService(id: string, props?: BaseServiceProps): Service
}
```

Modify the `fromHttpNamespaceAttributes` to return an object with it implemented:

```typescript
  /**
   * Creates a HTTP Namespace via all attributes.
   *
   * Note: While all attributes are required, only namespaceId is used (currently)
   *  by Service and so the other attributes may be set to placeholder values.
   */
  public static fromHttpNamespaceAttributes(scope: Construct, id: string, attrs: HttpNamespaceAttributes): IHttpNamespace {
    class Import extends Resource implements IHttpNamespace {
      public namespaceName = attrs.namespaceName;
      public namespaceId = attrs.namespaceId;
      public namespaceArn = attrs.namespaceArn;
      public type = NamespaceType.HTTP;

      public createService(_id: string, props?: BaseServiceProps): Service {
        return new Service(this, _id, {
          namespace: this,
          ...props,
        });
      }
    }
    return new Import(scope, id);
  }
```
 

### CDK CLI Version

2.23.0 (build 50444aa)

### Node.js Version

v18.4.0

### OS

All

### Language

Typescript

**Comment by rwxdash** [comment]
Similarly, one step further, `IService` that returns from `Service.fromServiceAttributes` doesn't include the `register*` methods. I just wanted to add as these issues are closely related.

**Comment by indrora** [comment]
This is a limitation of CloudFormation. When importing, you get an IThing because it's a cross-stack reference to another object that might not actually exist yet. 

Not a bug. Closing.

**Comment by Danno040** [comment]
I don't understand how @indrora 's comment results in this not being a bug (or at least a missing feature).

I thought the purpose of importing was to allow you to use non-CDK managed resources in your CDK code. This makes it the user's problem to ensure it exists, not CloudFormation.

I think I understand the logic of returning something that implements the interface (and not an instance of the usual class). However, I don't understand why that justifies having an _incomplete_ interface. There are [other instances](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L295) in the CDK where imported resources' interfaces include methods, so why is this an exception? Isn't the purpose of a Namespace to attach services to it? (Why else would I want to import it?)

I doubt it counts for munch, but I have been using the proposed fix in my own code since I opened the issue, and haven't had any problems.

**Comment by MrArnoldPalmer** [comment]
@Danno040 In this case, meaning `HttpNamespace.createService` and `Service.registerXxx`, the methods themselves create new resources and don't mutate the properties of the corresponding parent CFN resource in any way, so I will reopen this.

This is not the case for any number of methods on L2 constructs though which makes them incompatible with calling against the construct interface.

**Comment by indrora** [comment]
Yeah, I jumped the gun here a bit. It'd be nice if `IXxx` had a `CanMutate()` function that told you if the imported resource was mutable in context, but I don't know if we have that information here.

**Comment by Danno040** [comment]
Would it be helpful if I submitted a PR with the proposed change? Are there other things that need further consideration before implementation?
